README.md

```diff
@@ -507,34 +567,82 @@ val toolbarCustomization = ToolbarCustomizationBuilder()
-    super.onActivityResult(requestCode, resultCode, data)
-    if (requestCode == AUTHORIZING_PAYMENT_REQUEST_CODE && resultCode == RESULT_OK) {
-        val url = data?.getStringExtra(AuthorizingPaymentURLVerifier.EXTRA_RETURNED_URLSTRING)
-        // Use the redirected URL here
+ fun handleActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+
+    // custom result code when web view is closed
+    if (resultCode == AuthorizingPaymentActivity.WEBVIEW_CLOSED_RESULT_CODE) {
+        snackbar.setText(R.string.webview_closed).show()
+        return
+    }
+
+    if (resultCode == RESULT_CANCELED) {
+        snackbar.setText(R.string.payment_cancelled).show()
+        return
+    }
+
+    if (data == null) {
+        snackbar.setText(R.string.payment_success_but_no_result).show()
+        return
+    }
+
+    when (requestCode) {
+        AUTHORIZING_PAYMENT_REQUEST_CODE -> {
+            with(data.parcelable<AuthorizingPaymentResult>(AuthorizingPaymentActivity.EXTRA_AUTHORIZING_PAYMENT_RESULT)) {
+                Log.d(TAG, this.toString())
+                val resultMessage = when (this) {
+                    is AuthorizingPaymentResult.ThreeDS1Completed -> "Authorization with 3D Secure version 1 completed: returnedUrl=${returnedUrl}"
+                    is AuthorizingPaymentResult.ThreeDS2Completed -> "Authorization with 3D Secure version 2 completed: transStatus=${transStatus}"
+                    is AuthorizingPaymentResult.Failure -> {
+                        Log.e(TAG, throwable.message, throwable.cause)
+                        throwable.message ?: "Unknown error."
+                    }
+
+                    null -> "Not found the authorization result."
+                }
+                Log.d(TAG, resultMessage)
+                snackbar.setText(resultMessage).show()
+            }
+        }
     }
 }
 ```
+You can check out the sample implementation in the [CheckoutActivity](./app/src/kotlin/java/co/omise/android/example/CheckoutActivity.kt) class in the sample app.
 
-### Authorizing payment via an external app
+### Observing charge status in the token
 
-Some request methods allow the user to authorize the payment with an external app, for example Alipay. When a user needs to authorize the payment with an external app, `AuthorizingPaymentActivity` will automatically open an external app. However merchant developers must handle the `Intent` callback by themselves.
+This is a utility function for observing the token until its charge status changes. You can use this function for checking the charge status after the payment authorization process completes.
+
+```kotlin
+val client = Client("pkey_test_1234")
+client.observeTokenUntilChargeStatusChanged("tokn_test_1234", object: RequestListener<Token> {
+    override fun onRequestSucceed(model: Token) {
+        TODO("Not yet implemented")
+    }
 
-### 3D Secure 2
+    override fun onRequestFailed(throwable: Throwable) {
+        TODO("Not yet implemented")
```

Comment:
Are we going to leave it like this? 
`TODO("Not yet implemented")`

Reply:
@annatotest Yes the merchant will handle this on his side like show an error message for the user.